pkg/pool-weighted/contracts/LBPool.sol

```diff
@@ -0,0 +1,294 @@
+        // Ensure the caller is the owner, as only the owner can add liquidity.
+        hookFlags.shouldCallBeforeAddLiquidity = true;
+    }
+
+    /**
+     * @notice Check that the caller who initiated the add liquidity operation is the owner.
+     * @dev We first ensure the caller is the standard router, so that we know we can trust the value it returns
+     * from `getSender`.
+     *
+     * @param router The address (usually a router contract) that initiated the add liquidity operation
+     */
+    function onBeforeAddLiquidity(
+        address router,
+        address,
+        AddLiquidityKind,
+        uint256[] memory,
+        uint256,
+        uint256[] memory,
+        bytes memory
+    ) public view override onlyVault returns (bool) {
+        if (router == _TRUSTED_ROUTER) {
+            return IRouterCommon(router).getSender() == owner();
+        }
+
+        revert RouterNotTrusted();
+    }
+
+    /**
+     * @notice Called before a swap to let the pool block swaps if not enabled.
+     * @return success True if the pool has swaps enabled.
+     */
+    function onBeforeSwap(PoolSwapParams calldata, address) public view override onlyVault returns (bool) {
+        return _getPoolSwapEnabled();
+    }
+
+    /*******************************************************************************
+                                  Internal Functions
+    *******************************************************************************/
+
+    // This is unused in this contract, but must be overridden from WeightedPool for consistency.
+    function _getNormalizedWeight(uint256 tokenIndex) internal view virtual override returns (uint256) {
+        if (tokenIndex < _NUM_TOKENS) {
+            return _getNormalizedWeights()[tokenIndex];
+        }
+
+        revert IVaultErrors.InvalidToken();
+    }
+
+    function _getNormalizedWeights() internal view override returns (uint256[] memory) {
+        uint256[] memory normalizedWeights = new uint256[](_NUM_TOKENS);
+        normalizedWeights[0] = _getNormalizedWeight0();
+        normalizedWeights[1] = FixedPoint.ONE - normalizedWeights[0];
+
+        return normalizedWeights;
+    }
+
+    function _getNormalizedWeight0() internal view virtual returns (uint256) {
+        PoolState memory poolState = _poolState;
+        uint256 pctProgress = GradualValueChange.calculateValueChangeProgress(poolState.startTime, poolState.endTime);
+
+        return GradualValueChange.interpolateValue(poolState.startWeight0, poolState.endWeight0, pctProgress);
+    }
+
+    function _getPoolSwapEnabled() private view returns (bool) {
+        return _poolState.swapEnabled;
+    }
+
+    function _setSwapEnabled(bool swapEnabled) private {
+        _poolState.swapEnabled = swapEnabled;
+
+        emit SwapEnabledSet(swapEnabled);
+    }
+
+    /**
+     * @dev When calling updateWeightsGradually again during an update, reset the start weights to the current weights,
```

Comment:
The comment here is true, but it doesn't describe what the function is doing. Anything can be passed to this function as `startWeights`, it is not necessarily the current weights. This comment is more appropriate at line 169.

Reply:
I think this comment to less intended to describe the function's behavior and more to direct any developer how to use it. FWIW, this was a [direct copy/paste from v2](https://github.com/balancer/balancer-v2-monorepo/blob/master/pkg/pool-weighted/contracts/lbp/LiquidityBootstrappingPoolSettings.sol#L280-L281)